services/server/src/lib/assets.ts

```diff
@@ -19,16 +21,30 @@ export async function upload({
   data,
   key,
   type,
+  prefix = "",
 }: {
   data: Buffer;
   type?: string;
   key?: string | null;
+  prefix?: string;
 }) {
   const id = key || uuid();
-  await s3.put({ key: id, data, type });
+  await s3.put({ key: `${prefix}${id}`, data, type });
   return id;
 }
 
 export const uploadMiddleware = multer({
   storage: multer.memoryStorage(),
 });
+
+/**
+ * return null in case of unexpected error
+ */
+export function exceedsSizeLimit(size: number): boolean | null {
+  const limit = bytes("8mb");
+  if (!limit) {
+    logger().error("exceedsSizeLimit: an error occured with bytes function.");
+    return null;
+  }
+  return size > limit;
+}
```

Comment:
Not all files will have a max size of 8mb. Let's update this function look as follow: 

```suggestion
export function exceedsSizeLimit(size: number, max: "${number}mb"): boolean | null {
  const limit = bytes("8mb");
  if (!limit) throw new Error(`invalid size: ${max}`);
  return size > limit;
}
```